app/uk/gov/hmrc/cdsreimbursementclaimfrontend/controllers/claims/SelectDutiesController.scala

```diff
@@ -63,15 +68,17 @@ class SelectDutiesController @Inject() (
   def selectDuties(): Action[AnyContent] =
     authenticatedActionWithSessionData.async { implicit request =>
       withAnswers[DutiesSelectedAnswer] { (fillingOutClaim, previousAnswer) =>
-        getAvailableDuties(fillingOutClaim).fold(
+        val cmaEligibleDutiesMap: CmaEligibleAndDuties = getAvailableDuties(fillingOutClaim)
+
+        cmaEligibleDutiesMap.dutiesSelectedAnswer.fold(
           error => {
             logger.warn("No Available duties: ", error)
             Redirect(baseRoutes.IneligibleController.ineligible())
           },
-          dutiesAvailable => {
+          (dutiesAvailable: DutiesSelectedAnswer) => {
```

Comment:
Is it really necessary?

Reply:
@aneksamun  I added the type annotation while working on this to get a clearer idea what type `dutiesAvailable` is and I generally leave it as good practice for two reasons: 
1. It's less work for the compiler as it won't have to  do the extra work of inferring a type
2. It generally makes for better readability.

That being said, It's usually a matter of preference so it's not absolutely necessary. I can remove it if you want.

Reply:
@aneksamun  Here's an example of a [best practice](https://nrinaudo.github.io/scala-best-practices/binary_compat/explicit_type_annotations.html) for type inferrence. 
Also, IDE might show a type but not everyone uses the same IDE and even with IntelliJ, I think the type only shows if you have the ultimate version. 

Like I said, this is usually a matter of preference and I'm happy to change it if it's really an issue. 
We can have further discussions about it later.

Reply:
```suggestion
          dutiesAvailable => {
```

Reply:
Robert, that best practice is in regards public members. We have it built in linker check, so if you create a public member with no type assigned it fails to compile. The variables within clojure or lambda has private scope. 
Thank you!